bus-mapping/src/evm/opcodes/returndatasize.rs

```diff
@@ -0,0 +1,112 @@
+use crate::{
+    circuit_input_builder::{CircuitInputStateRef, ExecStep},
+    operation::CallContextField,
+    Error,
+};
+
+use eth_types::GethExecStep;
+
+use super::Opcode;
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct Returndatasize;
+
+impl Opcode for Returndatasize {
+    fn gen_associated_ops(
+        state: &mut CircuitInputStateRef,
+        geth_steps: &[GethExecStep],
+    ) -> Result<Vec<ExecStep>, Error> {
+        let geth_step = &geth_steps[0];
+        let mut exec_step = state.new_step(geth_step)?;
+        let value = geth_steps[1].stack.last()?;
+        state.call_context_read(
+            &mut exec_step,
+            state.call()?.call_id,
+            CallContextField::LastCalleeReturnDataLength,
+            value,
+        );
+
+        state.stack_write(
+            &mut exec_step,
+            geth_step.stack.last_filled().map(|a| a - 1),
+            value,
+        )?;
+
+        Ok(vec![exec_step])
+    }
+}
+
+#[cfg(test)]
+mod returndatasize_tests {
+    use crate::{
+        circuit_input_builder::ExecState,
+        mock::BlockData,
+        operation::{CallContextField, CallContextOp, StackOp, RW},
+    };
+    use eth_types::{
+        bytecode,
+        evm_types::{OpcodeId, StackAddress},
+        geth_types::GethData,
+    };
+
+    use mock::test_ctx::{helpers::*, TestContext};
+    use pretty_assertions::assert_eq;
+
+    #[test]
+    fn returndatasize_opcode_impl() {
+        let code = bytecode! {
+            RETURNDATASIZE
+            STOP
+        };
```

Comment:
I think this could be a bit more elaborated. 
We should create a sub-context with either: [CALL](https://www.evm.codes/#F1), [CALLCODE](https://www.evm.codes/#F2), [DELEGATECALL](https://www.evm.codes/#F4) or [STATICCALL](https://www.evm.codes/#FA) and then get a `RETURNDATASIZE` from there. I'd also try to play with really big values and values like 0 or similar. So that we test all the possible edge-cases.

Reply:
Thanks. I'm working on it.

Reply:
Updated the tests. Create a sub-context with ```CALL``` and then get a ```RETURNDATASIZE``` from there.